pkg/prometheus/server/operator.go

```diff
@@ -959,6 +964,11 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 	}
 
 	p := pobj.(*monitoringv1.Prometheus)
+	operator.UpdateObject(
```

Comment:
I don't think the operator should enforce the annotations on CRs, it's up to the user to provide CR's with or without the annotation

Reply:
I assume that would also depend on the implementation details, if we aim to create a random controller-id for any new one then this should be kept as-is.

Reply:
right, done